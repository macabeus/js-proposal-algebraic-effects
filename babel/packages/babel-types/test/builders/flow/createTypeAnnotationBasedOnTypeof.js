import { createTypeAnnotationBasedOnTypeof } from "../../..";

describe("builders", function() {
  describe("flow", function() {
    describe("createTypeAnnotationBasedOnTypeof", function() {
      const values = {
        string: typeof "string",
        number: typeof 123,
        true: typeof true,
        object: typeof {},
        undefined: typeof undefined,
        function: typeof function() {},
        symbol: typeof Symbol(),
      };

      for (const name in values) {
        const value = values[name];
        it(name, function() {
          const result = createTypeAnnotationBasedOnTypeof(value);
          expect(result).toMatchSnapshot();
        });
      }

      it("invalid", function() {
        expect(() =>
          createTypeAnnotationBasedOnTypeof("thisdoesnotexist"),
        ).toThrow(Error);
      });
    });
  });
});
