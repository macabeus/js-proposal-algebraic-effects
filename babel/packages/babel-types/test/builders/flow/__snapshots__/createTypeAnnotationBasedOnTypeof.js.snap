// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`builders flow createTypeAnnotationBasedOnTypeof function 1`] = `
Object {
  "id": Object {
    "name": "Function",
    "type": "Identifier",
  },
  "type": "GenericTypeAnnotation",
  "typeParameters": null,
}
`;

exports[`builders flow createTypeAnnotationBasedOnTypeof number 1`] = `
Object {
  "type": "NumberTypeAnnotation",
}
`;

exports[`builders flow createTypeAnnotationBasedOnTypeof object 1`] = `
Object {
  "id": Object {
    "name": "Object",
    "type": "Identifier",
  },
  "type": "GenericTypeAnnotation",
  "typeParameters": null,
}
`;

exports[`builders flow createTypeAnnotationBasedOnTypeof string 1`] = `
Object {
  "type": "StringTypeAnnotation",
}
`;

exports[`builders flow createTypeAnnotationBasedOnTypeof symbol 1`] = `
Object {
  "id": Object {
    "name": "Symbol",
    "type": "Identifier",
  },
  "type": "GenericTypeAnnotation",
  "typeParameters": null,
}
`;

exports[`builders flow createTypeAnnotationBasedOnTypeof true 1`] = `
Object {
  "type": "BooleanTypeAnnotation",
}
`;

exports[`builders flow createTypeAnnotationBasedOnTypeof undefined 1`] = `
Object {
  "type": "VoidTypeAnnotation",
}
`;
